aptos-move/aptos-release-builder/src/main.rs

```diff
@@ -51,21 +71,59 @@ async fn main() -> Result<()> {
         },
         Commands::ValidateProposals {
             release_config,
-            test_dir,
+            input_option,
             endpoint,
             framework_git_rev,
+            mint_to_validator,
         } => {
             let config =
                 aptos_release_builder::ReleaseConfig::load_config(release_config.as_path())?;
 
-            let mut network_config = aptos_release_builder::validate::NetworkConfig::new_from_dir(
-                endpoint,
-                test_dir.as_path(),
-            )?;
+            let root_key_path = aptos_temppath::TempPath::new();
+            root_key_path.create_as_file().unwrap();
+
+            let mut network_config = match input_option {
+                InputOptions::FromDirectory { test_dir } => {
+                    aptos_release_builder::validate::NetworkConfig::new_from_dir(
+                        endpoint,
+                        test_dir.as_path(),
+                    )?
+                },
+                InputOptions::FromArgs {
+                    root_key,
+                    validator_address,
+                    validator_key,
+                } => {
+                    let root_key = Ed25519PrivateKey::from_encoded_string(&root_key)?;
+                    let validator_key = Ed25519PrivateKey::from_encoded_string(&validator_key)?;
+                    let validator_account = AccountAddress::from_hex(validator_address.as_bytes())?;
+
+                    let mut root_key_path = root_key_path.path().to_path_buf();
+                    root_key_path.set_extension("key");
+
+                    std::fs::write(root_key_path.as_path(), bcs::to_bytes(&root_key)?)?;
+
+                    aptos_release_builder::validate::NetworkConfig {
+                        root_key_path,
+                        validator_account,
+                        validator_key,
+                        framework_git_rev: None,
+                        endpoint,
+                    }
+                },
+            };
 
             network_config.framework_git_rev = framework_git_rev;
 
-            aptos_release_builder::validate::validate_config(config, network_config).await
+            if mint_to_validator {
+                network_config.mint_to_validator().await?;
+            }
+
+            network_config.set_fast_resolve(30).await?;
```

Comment:
does this actually exist on devnet? (Thinking about how we're gonna execute today)

Reply:
This would exist.